File: Lab_10/BankApp/Commands/CreateAccountCommand.cs
using System.Threading.Channels;
using BankApp.App;

namespace BankApp.Commands;

public class CreateAccountCommand : ICommand
{
    private Bank _bank;
    private string _accountNumber;
    private decimal _initialDeposit;

    public CreateAccountCommand(Bank bank, string accountNumber, decimal initialDeposit)
    {
        _bank = bank;
        _accountNumber = accountNumber;
        _initialDeposit = initialDeposit;
    }

    public void Execute()
    {
        _bank.CreateAccount(_accountNumber, _initialDeposit);
    }

    public void Signature()
    {
        Console.WriteLine($"Created Bank Account at {_accountNumber} with diposit {_initialDeposit}");
    }
}

File: lab10/Bank Account Command Pattern/BankApp/src/Command/CreateAccountCommand.java
package Command;

import App.bank;

public class CreateAccountCommand implements Command{

    private bank bank;
    private String accountNumber;
    private double initialDeposit;

    public CreateAccountCommand(bank bank, String accountNumber, double initialDeposit) {
        this.bank = bank;
        this.accountNumber = accountNumber;
        this.initialDeposit = initialDeposit;
    }
    @Override
    public void execute() {
        bank.createAccount(accountNumber, initialDeposit);
    }

    @Override
    public void signature() {
        System.out.println("Created Bank Account at " + accountNumber + " with deposit " + initialDeposit);
    }
}
